homeassistant/components/owntracks/messages.py

```diff
@@ -39,11 +39,15 @@ def _parse_topic(topic, subscribe_topic):
     Async friendly.
     """
     subscription = subscribe_topic.split("/")
-    try:
-        user_index = subscription.index("#")
-    except ValueError:
+
+    user_index = None
+    for index, segment in enumerate(subscription):
+        if segment in ("#", "+"):
+            user_index = index
```

Comment:
This doesn't seem to work as expected. We should verify that there are two subsequent `+`. Otherwise someone might subscribe to `owntracks/my-user-id/+` and it would not work as expected.

Reply:
Thinking about this more, MQTT topic should not be used for feature selection.

Reply:
Can you elaborate what non-location events you're seeing as part of Home Assistant and why you want an option to filter those ?